mailbox/backup/src/main/java/org/apache/james/mailbox/backup/DefaultMailboxBackup.java

```diff
@@ -0,0 +1,129 @@
+        this.store = store;
+    }
+
+    private final MailboxManager mailboxManager;
+    private final Archive archive;
+    private final BlobStore store;
+
+    private Function<MailboxPath, Optional<MailAccountContent>> getMailboxWithAnnotationsFromPath(MailboxSession session) {
+        return path -> {
+            try {
+                StoreMessageManager messageManager = (StoreMessageManager) mailboxManager.getMailbox(path, session);
+                Mailbox mailbox = messageManager.getMailboxEntity();
+                List<MailboxAnnotation> annotations = mailboxManager.getAllAnnotations(path, session);
+                MailboxWithAnnotations mailboxWithAnnotations = new MailboxWithAnnotations(mailbox, annotations);
+                Stream<MessageResult> messages = Iterators.toStream(messageManager.getMessages(MessageRange.all(), FetchGroupImpl.FULL_CONTENT, session));
+                return Optional.of(new MailAccountContent(mailboxWithAnnotations, messages));
+            } catch (MailboxException e) {
+                LOGGER.error("Error while fetching Mailbox during backup", e);
+                return Optional.empty();
+            }
+        };
+    }
+
+    private List<MailAccountContent> getAllMailboxesForUser(MailboxSession session) throws MailboxException {
+        MailboxQuery queryUser = MailboxQuery.builder().username(session.getUser().asString()).build();
+        Stream<MailboxPath> paths = mailboxManager.search(queryUser, session).stream().map(mailboxMetaData -> mailboxMetaData.getPath());
+        List<MailAccountContent> mailboxes = paths
+            .flatMap(getMailboxWithAnnotationsFromPath(session).andThen(OptionalUtils::toStream))
+            .collect(Collectors.toList());
+
+        return mailboxes;
+    }
+
+    private Publisher<BlobId> saveToStore(List<MailboxWithAnnotations> mailboxes, Stream<MessageResult> messages) throws IOException {
+        File tmp = File.createTempFile(UUID.randomUUID().toString(), ".tmp");
+        try (OutputStream out = new FileOutputStream(tmp)) {
+            archive.archive(mailboxes, messages, out);
+            try (InputStream in = new BufferedInputStream(new FileInputStream(tmp))) {
+                return store.save(in).map(b -> BlobId.fromString(b.asString()));
+            }
+        } finally {
+            tmp.delete();
+        }
+    }
+
+    private Stream<MessageResult> allMessagesForUser(List<MailAccountContent> mailboxes) {
+        return mailboxes.stream().flatMap(messages -> messages.messages);
+    }
+
+    @Override
+    public Publisher<BlobId> backupAccount(User user) throws IOException, MailboxException {
+        MailboxSession session = mailboxManager.createSystemSession(user.asString());
+        List<MailAccountContent> mailboxesWithMessages = getAllMailboxesForUser(session);
+        List<MailboxWithAnnotations> mailboxes = mailboxesWithMessages.stream().map(m -> m.mailboxWithAnnotations).collect(Collectors.toList());
```

Comment:
`m` is not explicit enough (`message`)
And as @mbaechler said as well on an other comment, we use immutable lists to collect